lib/internal/timers.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+// Timeout values > TIMEOUT_MAX are set to 1.
+const TIMEOUT_MAX = 2147483647; // 2^31-1
+
+module.exports = {
+  TIMEOUT_MAX,
+  kTimeout: Symbol('timeout'), // For hiding Timeouts on other internals.
+  Timeout,
+  setUnrefTimeout
+};
+
+// Timer constructor function.
+// The entire prototype is defined in lib/timers.js
+function Timeout(after, callback, args) {
+  this._called = false;
+  this._idleTimeout = after;
+  this._idlePrev = this;
+  this._idleNext = this;
+  this._idleStart = null;
+  this._onTimeout = callback;
+  this._timerArgs = args;
+  this._repeat = null;
+}
+
+var timers;
+function getTimers() {
+  if (timers === undefined) {
+    timers = require('timers');
+  }
+  return timers;
+}
+
+function setUnrefTimeout(callback, msecs) {
+  if (typeof callback !== 'function') {
+    throw new TypeError('"callback" argument must be a function');
+  }
+
+  if (typeof msecs !== 'number') {
+    throw new TypeError('"msecs" argument must be a number, was ' + msecs);
+  }
+
+  if (msecs < 0 || !isFinite(msecs)) {
```

Comment:
Since we already know it's a number type, I think it may be faster to just inline the necessary checks:

```js
if (msecs < 0 || msecs !== msecs || msecs === Infinity) {
```